client/packages/coldchain/src/Monitoring/api/TemperatureBreach/hooks/document/useTemperatureBreaches.ts

```diff
@@ -1,12 +1,16 @@
-import { useUrlQueryParams } from '@common/hooks';
+import { FilterByWithBoolean, SortBy } from '@common/hooks';
 import { useTemperatureBreachApi } from '../utils/useTemperatureBreachesApi';
 import { useQuery } from '@openmsupply-client/common';
+import { TemperatureBreachFragment } from '../../operations.generated';
 
-export const useTemperatureBreaches = () => {
-  const { queryParams } = useUrlQueryParams({
-    initialSort: { key: 'startDatetime', dir: 'desc' },
-  });
+export type ListParams = {
+  first: number;
+  offset: number;
+  sortBy: SortBy<TemperatureBreachFragment>;
+  filterBy: FilterByWithBoolean | null;
+};
 
+export const useTemperatureBreaches = (queryParams: ListParams) => {
```

Comment:
lifted these out, there were errors being thrown due to this call being made when the URL had an unsupported sort key

Reply:
Good idea. This also make the hook much more flexible and IMHO also more readable. For example, I was initially very confused how these hooks know their query parameters.

For new API hooks I am also tending to keep them as close as possible to the underlying API call to avoid confusion like that. I then also use the original graphql filter type to get the exact filter options. What is actually the advantage of FilterByWithBoolean?

Reply:
The `FilterByWithBoolean` type is required because this filter has `acknowledged: boolean` as one of the options, and that doesn't fit the standard `FilterRule` syntax

Reply:
I more meant why not use the original API types?

Reply:
ah. I see - thanks, that's much nicer!